application/classes/Ushahidi/Repository/Form/Contact.php

```diff
@@ -110,19 +121,25 @@ public function updateCollection(Array $entities, $form_id = null)
 			 * in phone number validation but we don't want to save it
 			 */
 			unset($entity->country_code);
-			$query = DB::insert($this->getTable())
-				->columns(array_keys($entity->asArray()));
-			$query->values($entity->asArray());
-			$result = $query->execute($this->db);
-			if (!isset($result[0])) {
-				throw new HTTP_Exception_500(
-					sprintf(
-						'Could not create contacts. Result:  %s',
-						var_export($entity, true)
-					)
-				);
+			if (!$entity->id) {
+
+				$query = DB::insert($this->getTable())
```

Comment:
Not a blocker, but curious if we could refactor the insert into its own method, (if something similar doesn't already exist), since `updateCollection()` is looking a little large.